src/Lean/Compiler/IR/EmitLLVM.lean

```diff
@@ -1509,9 +1509,16 @@ def optimizeLLVMModule (mod : LLVM.Module ctx) : IO Unit := do
   LLVM.disposePassManager pm
   LLVM.disposePassManagerBuilder pmb
 
+/--
+`emitLLVM` is the entrypoint for the lean shell to code generate LLVM.
+TODO(bollu): tripleStr? has type `String` instead of `Option String`
+             because there didn't seem to be a convenient way to convert a
+             `lean::optional<string>` to the correct `object_ref` which represents
```

Comment:
https://github.com/leanprover/lean4/blob/67a58467429c9c47e40aef3d04aa50c71eacd2b5/src/runtime/object.h#L305-L306

Reply:
Ah, but also
https://github.com/leanprover/lean4/blob/67a58467429c9c47e40aef3d04aa50c71eacd2b5/src/runtime/object_ref.h#L171-L179
which should work with `optional<string_ref>`

Reply:
Hum, but I start with an `optional<std::string>`, not an `optional<string_ref>`. I elected to use the low technology solution and build the `lean_object` by hand.

Reply:
> Hum, but I start with an `optional<std::string>`, not an `optional<string_ref>`.

There's no reason you couldn't do the latter though.